File: src/main/java/web/veterinaria/entity/Cita.java
package web.veterinaria.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "cita")
public class Cita {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "IdCita")
    private Long idCita;

    @Column(name = "FechaHora", nullable = false)
    private LocalDateTime fechaHora;

    @Column(name = "Observaciones", length = 500)
    private String observaciones;

    @Column(name = "FechaRegistro", nullable = false)
    private LocalDateTime fechaRegistro = LocalDateTime.now();

    @ManyToOne
    @JoinColumn(name = "IdMascota", nullable = false)
    private Mascota mascota;

    @ManyToOne
    @JoinColumn(name = "IdVeterinario", nullable = false)
    private Veterinario veterinario;

    @ManyToOne
    @JoinColumn(name = "IdEstadoCita", nullable = false)
    private EstadoCita estadoCita;
}

File: proyecto-veterinari-frontend/src/app/models/cita.ts
export interface Cita {}

export interface CitaRequest {
  idMascota: number;
  idVeterinario: number;
  fechaHora: string; // ISO 8601: "2026-08-15T10:30:00"
  observaciones?: string;
  idEstadoCita?: number; // 1=Pendiente, 2=Confirmada, 3=Atendida, 4=Cancelada
}

export interface CitaResponse {
  idCita: number;
  idMascota: number;
  nombreMascota: string;
  idVeterinario: number;
  nombreVeterinario: string;
  fechaHora: string;
  estadoCita: string;
  observaciones: string;
  fechaRegistro: string;
}
